### PR TITLE
Add regex pattern to match the query output

### PR DESCRIPTION
**Pull request by jindraj**
## Pull Request Checklist

No, but I would welcome this change.

#### General

- [x] Update Changelog following the conventions laid out  [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### Purpose
Add ability to check query output with configured regular expression

### CODE REVIEW

**Comment by phumpal** [comment]
Interesting, I'm curious about use case. Can you provide an example?

**Comment by jindraj** [comment]
> Interesting, I'm curious about use case. Can you provide an example?

currently the check is looking only at the number of tuples returned. This PR is a way to check result string. 

We use it to check presence of a specific string or pattern in output of the query. We also have larger queries stored as functions which does a lot of things and returns "OK", or something else. I'm not author of queries executed by check and since they are quite complex and I'm not familiar with the database schema I can't give a better example.
But the simplest one:
```
check-postgres-query.rb -u sensu -p egcBVKDvQ_JKMaSMudZL -q  "select 'OK'" -r 'OK'
```

Maybe limiting checking just to res.getvalue(0, 0) isn't the best, but it works well for our use case. I'm open to make any changes to make this functionality better.

I believe this can be useful for more sensu-plugins-postgresql users.

**Comment by phumpal** [comment]
:+1: cool, makes sense. I wanted to make sure I understood the use case and be able to provide an adequate local test case.

Would you mind rebasing?

/cc @majormoses

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md).
 
 ## [Unreleased]
+### Update check-postgres-query.rb
```
Comment (majormoses) [inline CHANGELOG.md]:
```suggestion
### Added
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+- Add `-r`, `--regex-pattern` to match query result against (@jindraj)
```
Comment (majormoses) [inline CHANGELOG.md]:
```suggestion
- `check-postgres-query.rb`: Add `-r`, `--regex-pattern` to match query result against (@jindraj)
```

File: bin/check-postgres-query.rb [hunk 3]
```diff
@@ -73,6 +73,11 @@ class CheckPostgresQuery < Sensu::Plugin::Check::CLI
          long: '--query QUERY',
          required: true
 
+  option :regex_pattern,
+         description: 'Query output regex pattern',
```
Comment (majormoses) [inline bin/check-postgres-query.rb]:
```suggestion
         description: 'Regex pattern to match on query results and alert on if it does not match',
```

**Review by majormoses — COMMENTED** [review]
Makes sense to me, couple comments/suggestions. Great work!

**Comment by jindraj** [comment]
Sorry folks, I totally messed up. Git isn't my strongest skill.
I created a new PR instead: https://github.com/sensu-plugins/sensu-plugins-postgres/pull/121

**Pull request closed without merging**